pkg/crinit/aggregated/aggregated_test.go

```diff
@@ -0,0 +1,515 @@
+			errExpected: false,
+		},
+		{
+			desc:        "other service type not supported",
+			initialOpts: &aggregatedClusterRegistryOptions{apiServerServiceTypeString: string(v1.ServiceTypeExternalName)},
+			errExpected: true,
+		},
+		{
+			desc: "advertise address supported with NodePort service type",
+			initialOpts: &aggregatedClusterRegistryOptions{
+				apiServerServiceTypeString: string(v1.ServiceTypeNodePort),
+				SubcommandOptions: util.SubcommandOptions{
+					APIServerAdvertiseAddress: "10.0.0.1"}},
+			errExpected: false,
+		},
+		{
+			desc: "advertise address not supported with non-NodePort service type",
+			initialOpts: &aggregatedClusterRegistryOptions{
+				apiServerServiceTypeString: string(v1.ServiceTypeLoadBalancer),
+				SubcommandOptions: util.SubcommandOptions{
+					APIServerAdvertiseAddress: "10.0.0.1"}},
+			errExpected: true,
+		},
+		{
+			desc: "advertise address validated",
+			initialOpts: &aggregatedClusterRegistryOptions{
+				apiServerServiceTypeString: string(v1.ServiceTypeNodePort),
+				SubcommandOptions: util.SubcommandOptions{
+					APIServerAdvertiseAddress: "notAValidIP"}},
+			errExpected: true,
+		},
+		{
+			desc: "advertise port supported with NodePort service type",
+			initialOpts: &aggregatedClusterRegistryOptions{
+				apiServerServiceTypeString: string(v1.ServiceTypeNodePort),
+				SubcommandOptions: util.SubcommandOptions{
+					APIServerNodePortPort: testNodePort}},
+			finalOpts: &aggregatedClusterRegistryOptions{
+				apiServerServiceTypeString: string(v1.ServiceTypeNodePort),
+				SubcommandOptions: util.SubcommandOptions{
+					APIServerServiceType:     v1.ServiceTypeNodePort,
+					APIServerNodePortPort:    testNodePort,
+					APIServerNodePortPortPtr: &testNodePort}},
+			errExpected: false,
+		},
+		{
+			desc: "advertise port rejected with non-NodePort service type",
+			initialOpts: &aggregatedClusterRegistryOptions{
+				apiServerServiceTypeString: string(v1.ServiceTypeLoadBalancer),
+				SubcommandOptions: util.SubcommandOptions{
+					APIServerNodePortPort: testNodePort}},
+			errExpected: true,
+		},
+		{
+			desc: "advertise port rejected if out of range",
+			initialOpts: &aggregatedClusterRegistryOptions{
+				apiServerServiceTypeString: string(v1.ServiceTypeNodePort),
+				SubcommandOptions: util.SubcommandOptions{
+					APIServerNodePortPort: 100000}},
+			errExpected: true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			err := validateOptions(tc.initialOpts)
+			if tc.errExpected {
+				if err == nil {
+					t.Error("expected error, but no error was returned")
+				}
+			} else {
+				if err != nil {
+					t.Errorf("no error expected, but got %v", err)
+				}
+				if tc.finalOpts != nil {
+					if !reflect.DeepEqual(tc.initialOpts, tc.finalOpts) {
+						t.Errorf("unexpected output: got: %v, want: %v", tc.initialOpts, tc.finalOpts)
+					}
+				}
+			}
+		})
+	}
+}
+
+func TestMarshalOptions(t *testing.T) {
```

Comment:
Is the intention to migrate this to a shared test later? `MarshalOptions` isn't in this package.